geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -72,69 +70,73 @@ public LocatorClusterManagementService(DistributionManager distributionManager,
 
   @Override
   public ClusterManagementResult create(CacheElement config, String group) {
+    final ClusterManagementResultFactory cmrFactory = new ClusterManagementResultFactory();
+
     if (group == null) {
       group = "cluster";
     }
 
     if (persistenceService == null) {
-      return new ClusterManagementResult(false,
-          "Cluster configuration service needs to be enabled");
+      return cmrFactory.setPersistenceStatusNA("Cluster configuration service needs to be enabled")
+          .createResult();
     }
 
-    ClusterManagementResult result = new ClusterManagementResult();
     ConfigurationMutator configurationMutator = mutators.get(config.getClass());
 
     ConfigurationValidator validator = validators.get(config.getClass());
     if (validator != null) {
       try {
         validator.validate(config);
       } catch (IllegalArgumentException e) {
-        return new ClusterManagementResult(false, e.getMessage());
+        return cmrFactory.setPersistenceStatus(false, e.getMessage()).createResult();
       }
     }
 
     // exit early if config element already exists in cache config
-    CacheConfig currentPersistedConfig = persistenceService.getCacheConfig(group, true);
-    if (configurationMutator.exists(config, currentPersistedConfig)) {
-      throw new EntityExistsException("cache element " + config.getId() + " already exists.");
+    if (configurationMutator.exists(config, persistenceService.getCacheConfig(group, true))) {
+      return cmrFactory.setPersistenceStatus(Status.Result.NO_OP,
+          "cache element " + config.getId() + " already exists.").createResult();
     }
```

Comment:
I'm coming around to the way of thinking that this change should be reverted because it is "business logic" that should happen in specific `Controller` classes where it makes sense. The logic here in `LocatorClusterManagementService` should be very general, limited to detecting the existence of a config element already in the `CacheConfig`.